main/options/options.cc

```diff
@@ -780,8 +780,6 @@ void readOptions(Options &opts,
 
         bool enableAllLSPFeatures = raw["enable-all-experimental-lsp-features"].as<bool>();
         opts.lspAllBetaFeaturesEnabled = enableAllLSPFeatures || raw["enable-all-beta-lsp-features"].as<bool>();
-        opts.lspDocumentSymbolEnabled =
-            opts.lspAllBetaFeaturesEnabled || raw["enable-experimental-lsp-document-symbol"].as<bool>();
```

Comment:
I haven't removed the option itself because I find it's always annoying to have to jump back and forth between Sorbet versions when the new version says "this option is unrecognized."

We can delete the option some time in the future after more time has passed.